bgpd/bgp_vty.c

```diff
@@ -3103,9 +3253,16 @@ DEFUN (no_bgp_default_show_hostname,
        "Configure BGP defaults\n"
        "Show hostname in certain command outputs\n")
 {
-	VTY_DECLVAR_CONTEXT(bgp, bgp);
-	UNSET_FLAG(bgp->flags, BGP_FLAG_SHOW_HOSTNAME);
-	return CMD_SUCCESS;
+	nb_cli_enqueue_change(vty, "./show-hostname", NB_OP_MODIFY, "false");
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+void cli_show_router_bgp_show_hostname(struct vty *vty, struct lyd_node *dnode,
+				       bool show_defaults)
+{
+	if (yang_dnode_get_bool(dnode, NULL) != SAVE_BGP_SHOW_HOSTNAME)
```

Comment:
This isn't enough to handle the CLI configuration profiles properly, as whether the NB calls this callback or not depends solely on the default value specified in the YANG module.

To fix this problem we probably need the new `cli_show_ext()` callback proposed by this commit: https://github.com/opensourcerouting/frr/commit/d006e3d94f0941d2f6e0b7ff45b9198911f9d8f3

Reply:
Ack. This is a first set of changes to handle cli_show. Once the changes in opensourcerouting@d006e3d in subsequent PR can modify/accomodate to to `cli_show_ext() based on the (loaded) profile.

Reply:
Works for me.